include/wil/registry_helpers.h

```diff
@@ -872,41 +919,40 @@ namespace wil
                 template <typename R>
                 typename err_policy::result set_value_with_type(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const R& value, DWORD type) const
                 {
-                    const auto error = ::RegSetKeyValueW(
-                        m_key,
-                        subkey,
-                        value_name,
-                        type,
-                        static_cast<BYTE*>(reg_value_type_info::get_buffer(value)),
-                        reg_value_type_info::get_buffer_size(value));
-                    return err_policy::HResult(HRESULT_FROM_WIN32(error));
+                    return err_policy::HResult(HRESULT_FROM_WIN32(
+                        ::RegSetKeyValueW(
+                            m_key,
+                            subkey,
+                            value_name,
+                            type,
+                            static_cast<BYTE*>(reg_value_type_info::get_buffer(value)),
+                            reg_value_type_info::get_buffer_size_bytes(value))));
                 }
 
                 template <typename R, typename get_value_with_type_policy = err_policy>
                 typename get_value_with_type_policy::result get_value_with_type(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, R& return_value, DWORD type = reg_value_type_info::get_value_type<R>()) const
                 {
-                    HRESULT hr = S_OK;
-
                     if
 #if defined(__cpp_if_constexpr)
                         constexpr
 #endif
                         (reg_value_type_info::supports_prepare_buffer<R>())
 
                     {
-                        hr = reg_value_type_info::prepare_buffer(return_value);
-                        if (FAILED(hr))
+                        const auto prepare_buffer_error = reg_value_type_info::prepare_buffer(return_value);

```

Comment:
Is this a win32 error or an HRESULT? `prepare_buffer_hr`?

Reply:
Sure. renamed to _hr (and the other similar variables)